File: crates/iota-sdk/examples/generate_mnemonic.rs
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use iota_crypto::mnemonic::{MnemonicWordCount, generate_mnemonic};

fn main() {
    let mnemonic = generate_mnemonic(None);
    println!("24 word mnemonic: {mnemonic}");
    let mnemonic = generate_mnemonic(MnemonicWordCount::Words12);
    println!("12 word mnemonic: {mnemonic}");
}

File: bindings/wasm/examples/generate_mnemonic.mjs
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { generateMnemonic, MnemonicLength, initAsync } from "@iota/sdk-wasm";

await initAsync();

console.log("24 word mnemonic:", generateMnemonic(undefined));
console.log("12 word mnemonic:", generateMnemonic(MnemonicLength.Words12));
